core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -245,16 +282,20 @@ public void setRetryPolicy(RetryPolicy retryPolicy) {
             retryPolicy(retryPolicy);
         }
 
+        @Override
+        public RetryPolicy retryPolicy() {
+            return retryPolicy;
+        }
+
         @Override
         public Builder executionInterceptors(List<ExecutionInterceptor> executionInterceptors) {
-            this.executionInterceptors.clear();
-            this.executionInterceptors.addAll(executionInterceptors);
+            this.executionInterceptors = Collections.unmodifiableList(executionInterceptors);
             return this;
         }
 
         @Override
-        public Builder addExecutionInterceptor(ExecutionInterceptor executionInterceptors) {
-            this.executionInterceptors.add(executionInterceptors);
+        public Builder addExecutionInterceptor(ExecutionInterceptor executionInterceptor) {
+            this.executionInterceptors.add(executionInterceptor);
```

Comment:
This seems like it would be broken as you can't add to an unmodifiable collection. I.e. if you set executionInterceptors and then call add this should fail.